Use setting to determine if test should be ran for full installation

chadwilken opened the pull request:
There was some discussion about bypassing the `bundle install test` command during installation in #95. I added an optional setting `bundle_check_before_install` that defaults to `true`. If you manually set to false it will perform a full installation with every deploy.

chadwilken commented:
I resolved the merge conflicts, let me know if there is something else you need added to this.

will-in-wi commented on lib/capistrano/tasks/bundler.cap:
```diff
@@ -28,7 +28,8 @@ namespace :bundler do
           options << "--path #{fetch(:bundle_path)}" if fetch(:bundle_path)
-          if test(:bundle, :check, *options)
+
+          unless fetch(:bundle_check_before_install) && test(:bundle, :check, *options)
```
I'm not sure this is correct…
```ruby
unless fetch(:bundle_check_before_install) && test(:bundle, :check, *options)
```
which is the same as:
```ruby
if !(fetch(:bundle_check_before_install) && test(:bundle, :check, *options))
```
Which (applying DeMorgan's laws) is the same as
```ruby
if !fetch(:bundle_check_before_install) || !test(:bundle, :check, *options)
```

So if the `bundle_check_before_install` flag is false or `bundle check` returns false, we decide the dependencies are satisfied and skip installation.

Should it be:

```suggestion
          if fetch(:bundle_check_before_install) && test(:bundle, :check, *options)
```

where if the flag is true and bundle check returns true, we skip installation. If either of those are false, we run the installation.

Am I reading this correctly?

chadwilken replied:
I think that would be correct. I apologize, I haven't messed with this in quite a while so I am a bit rusty on all of the details.

chadwilken commented:
@will-in-wi I added documentation to the README and updated the logic. You're correct about the logic for the `if/unless` statement. I was incorrectly handling the situation when they want to run `bundle check` to decide if `bundle install` is necessary.

will-in-wi commented:
Thanks!
